exampleCourse/serverFilesCourse/r_autograder/pltest.R

```diff
@@ -0,0 +1,171 @@
+## Script Requirements ----
+
+# Note: Packages listed here are found on the PrairieLearn R docker image
+
+# Use testthat to test the student code
+library("testthat")
+
+# Use stringr to simplify string manipulation
+library("stringr")
+
+# Use jsonlite to export the test results under a PrairieLearn format
+library("jsonlite")
+
+## Script hierarchy
+# /grade
+# |- data/
+# |- results/
+# |- run/                # <--- Working directory
+# |- serverFilesCourse/
+# |- shared/
+# |- student/
+# |- tests/
+
+## Run Unit Tests ----
+
+# Automatically test the directory
+# This helps avoid needing to build a package on a per question basis.
+# To understand testing outside of a package, it's mainly documented in an
+# issue ticket: https://github.com/r-lib/testthat/issues/659
+
+# Begin the process of grading each student's code
+# Add a capture statement to avoid allowing students to output
+# content to console.
+log <- capture.output({
+    test_data_as_list <- testthat::test_dir("../tests",
+                                            reporter = testthat::ListReporter)
+})
+
+# Convert from list to a data.frame
+test_data = as.data.frame(test_data_as_list)
+
+## Compute Unit Test Correctness ----
+
+# Retrieve question totals from the test description. To facilitate this,
+# we assume the format is given as...
+#
+# NumericPoints: test description
+#
+# To ensure no issue, we remove all none-numeric values on the left and right
+# of the colon separator.
+
+# Obtain the maximum points
+test_data$max_points = stringr::str_replace(
+    test_data$test,
+    pattern = ".*([:digit:]+).*:.*",
+    replacement = "\\1"
+)
+
+# Convert retrieved maximum points to a numeric value
+test_data$max_points = as.numeric(test_data$max_points)
+
+# Calculate the number of points the student received
+test_data$earned_points = ifelse(test_data$passed >= 1,
+                                 test_data$max_points,
+                                 0.0)
+
+# Now, let's aim to extract the description values
+test_data$description = stringr::str_replace(
+    test_data$test,
+    pattern = ".*[:digit:]+.*:[:space:](.*)",
+    replacement = "\\1"
+)
+
+## Format the Results to Match External Grading Format ----
+
+# Format the results of each inline test.
+pl_test_list_format = function(i, x) {
+    
+    pl_message_statement = function(x) {
+        if (isTRUE(x$error)) {
+            test_stack = unlist(x$result, recursive = FALSE)
+            error_message = if (!is.null(test_stack) && length(test_stack) != 0) {
+                format(test_stack[[1]])
+            } else {
+                "Error message was unrecoverable."
```

Comment:
I’m getting this “unrecoverable” error message on any syntax error in the code. For example, leaving off the closing brace in the function, or having `retrn` rather than `return`. Can we display the actual error message to the student in this case? I would have expect some sort of stack trace with `Syntax error: unknown identifier “retrn”` or whatever.